### ISSUE TITLE
@truffle/fetch-and-compile doesn't report which fetcher it used

### ISSUE THREAD

**Issue by gnidan** [trufflesuite/truffle#4446]
## Issue

This package supports fetching from both Etherscan and from Sourcify, but it does not provide any output to say which of these was used. This information is a **must** for certain use cases, where you want to be dang-sure that you can verify a contract instance.

## Steps to Reproduce

1. Run this:
    ```typescript
    import { fetchAndCompile } from "@truffle/fetch-and-compile";
    /* ... */
    const { 
      fetchedFrom // (or another name)
    } = await fetchAndCompile(/* ... */);
    ``` 
2. Observe that there is no such output.

## Expected Behavior

I'd expect to find this information.

## Actual Results

There is no such output.

## Environment

* Operating System: 
* Ethereum client:
* Truffle version (`truffle version`): v5.4.21
* node version (`node --version`):
* npm version (`npm --version`):

**Comment by haltman-at** [comment]
To be clear, this issue is only about the `fetchAndCompile` function in specific, rather than the package as a whole, right?  Like I don't think we need to make the debugger version do this, right?  Assuming it's just the former, I think this should be pretty straightforward.

**Comment by gnidan** [comment]
Yep!
